ray-operator/controllers/ray/networkpolicy_controller.go

```diff
@@ -0,0 +1,460 @@
+	operatorNamespace := os.Getenv("POD_NAMESPACE")
+	if operatorNamespace == "" {
+		operatorNamespace = "ray-system" // fallback
+	}
+	return []string{operatorNamespace}
+}
+
+// createOrUpdateNetworkPolicy creates or updates a NetworkPolicy
+func (r *NetworkPolicyController) createOrUpdateNetworkPolicy(ctx context.Context, instance *rayv1.RayCluster, networkPolicy *networkingv1.NetworkPolicy) error {
+	logger := ctrl.LoggerFrom(ctx).WithName("networkpolicy-controller")
+
+	// Set owner reference for garbage collection
+	if err := controllerutil.SetControllerReference(instance, networkPolicy, r.Scheme); err != nil {
+		return err
+	}
+
+	// Try to create the NetworkPolicy
+	if err := r.Create(ctx, networkPolicy); err != nil {
+		if errors.IsAlreadyExists(err) {
+			// NetworkPolicy exists, update it
+			existing := &networkingv1.NetworkPolicy{}
+			if err := r.Get(ctx, client.ObjectKeyFromObject(networkPolicy), existing); err != nil {
+				return err
+			}
+
+			// Update the existing NetworkPolicy
+			existing.Spec = networkPolicy.Spec
+			existing.Labels = networkPolicy.Labels
+
+			if err := r.Update(ctx, existing); err != nil {
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, string(utils.FailedToCreateNetworkPolicy),
+					"Failed to update NetworkPolicy %s/%s: %v", networkPolicy.Namespace, networkPolicy.Name, err)
+				return err
+			}
+
+			logger.Info("Successfully updated NetworkPolicy", "name", networkPolicy.Name)
+			r.Recorder.Eventf(instance, corev1.EventTypeNormal, string(utils.CreatedNetworkPolicy),
+				"Updated NetworkPolicy %s/%s", networkPolicy.Namespace, networkPolicy.Name)
+		} else {
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, string(utils.FailedToCreateNetworkPolicy),
+				"Failed to create NetworkPolicy %s/%s: %v", networkPolicy.Namespace, networkPolicy.Name, err)
+			return err
+		}
+	} else {
+		logger.Info("Successfully created NetworkPolicy", "name", networkPolicy.Name)
+		r.Recorder.Eventf(instance, corev1.EventTypeNormal, string(utils.CreatedNetworkPolicy),
+			"Created NetworkPolicy %s/%s", networkPolicy.Namespace, networkPolicy.Name)
+	}
+
+	return nil
+}
+
+// buildHeadNetworkPolicy creates a NetworkPolicy for Ray head pods
+func (r *NetworkPolicyController) buildHeadNetworkPolicy(instance *rayv1.RayCluster, kubeRayNamespaces []string) *networkingv1.NetworkPolicy {
+	labels := map[string]string{
+		utils.RayClusterLabelKey:                instance.Name,
+		utils.KubernetesApplicationNameLabelKey: utils.ApplicationName,
+		utils.KubernetesCreatedByLabelKey:       utils.ComponentName,
+	}
+
+	// Build secured ports - mTLS port always included
+	allSecuredPorts := []networkingv1.NetworkPolicyPort{
+		{
+			Protocol: &[]corev1.Protocol{corev1.ProtocolTCP}[0],
+			Port:     &[]intstr.IntOrString{intstr.FromInt(8443)}[0],
+		},
+	}
+
+	// Check if mTLS is enabled by looking for TLS configuration in RayCluster
+	if r.isMTLSEnabled(instance) {
```

Comment:
This needs to be removed for now until we add the changes to the spec where the user has options around selecting the various configuarations. For now it'll be covered by the annotation